packages/providers/google/src/base.ts

```diff
@@ -278,24 +279,26 @@ export class Gemini extends ToolCallLLM<GeminiAdditionalChatOptions> {
   ): GeminiChatStreamResponse {
     const context = getChatContext(params);
     const client = this.session.getGenerativeModel(this.metadata);
-    const chat = client.startChat(
-      params.tools
-        ? {
-            history: context.history,
-            tools: [
-              {
-                functionDeclarations: params.tools.map(
-                  mapBaseToolToGeminiFunctionDeclaration,
-                ),
-              },
-            ],
-            safetySettings: DEFAULT_SAFETY_SETTINGS,
-          }
-        : {
-            history: context.history,
-            safetySettings: DEFAULT_SAFETY_SETTINGS,
+    const tools = params.tools?.length
```

Comment:
Thanks @atpsito and @marcusschiesser  for the fix, I was also facing the problem.
I would suggest to make the same change in `nonStreamChat`?
https://github.com/run-llama/LlamaIndexTS/blob/main/packages/providers/google/src/base.ts#L240
Maybe `nonStreamChat` and `streamChat` could share a `createStartChatParams` utility function?

Reply:
yes that would be great, want to send a PR?

Reply:
Just created https://github.com/run-llama/LlamaIndexTS/pull/1716